### DISCUSSION TITLE
How to get the robot pose (x, y, z, roll, pitch, yaw) in world frame?

### DISCUSSION THREAD

**Discussion by DeepDuke** [cyberbotics/webots#6389] [General]
I want to get the ground truth pose (x, y, z, roll, pitch, yaw) for a robot in simulation. Could anyone tell me how to get it? It seems very hard to find the api?

**Comment by BenjaminDeleze** [comment]
You can get it with a supervisor: https://cyberbotics.com/doc/reference/supervisor#wb_supervisor_node_get_position

**Comment by FrogGuaGuaGua** [comment]
https://github.com/cyberbotics/webots/discussions/5387

**Comment by DeepDuke** [comment]
Thank you!
